### PR TITLE
unix, windows: make uv_cwd be consistent with uv_exepath

### LINKED ISSUE #446: uv_cwd and uv_exepath should have similar interface
Right now it's a mess.

### PR DESCRIPTION
**Pull request by saghul**
Fixes #446

### CODE REVIEW

**Comment by saghul** [comment]
@piscisaureus @indutny, please have a look guys.

**Comment by indutny** [comment]
De ja vu? LGTM

**Comment by saghul** [comment]
You saw the unix part attached to the issue, this is the complete version, with extra bacon :-)

**Comment by saghul** [comment]
@indutny can you please review the rest of it? Tests are passing here :-)

File: src/win/util.c [hunk 1]
```diff
@@ -192,19 +192,36 @@ int uv_cwd(char* buffer, size_t size) {
...
+                          0,
+                          utf16_buffer,
+                          -1,
+                          NULL,
+                          0,
+                          NULL,
+                          NULL);
+  if (r == 0) {
+    return uv_translate_sys_error(GetLastError());
+  } else if (r > (int)*size) {
```
Comment (indutny) [inline src/win/util.c]: `(int) *size`

**Comment by indutny** [comment]
LGTM, except one tiny nit.

**Comment by saghul** [comment]
Thanks Fedor! Fixed the nit and landed it.

FYI, @tjfontaine when you update libuv you'll need to adapt to this change in node. Should be a couple of lines.

**Comment by piscisaureus** [comment]
lgtm

**Pull request merged**